cmd/cody-gateway/internal/actor/dotcomuser/dotcomuser.go

```diff
@@ -176,6 +181,32 @@ func (s *Source) get(ctx context.Context, token string, bypassCache bool) (*acto
 	return act, nil
 }
 
+func updateWithGracePeriod(act *actor.Actor, fetch func() (*actor.Actor, error), updateTime time.Time) (*actor.Actor, error) {
+	if act.LastUpdated == nil || time.Since(*act.LastUpdated) > defaultUpdateInterval {
```

Comment:
nit: I think this would read a little easier if we did a fast pass at the beginning: 


```suggestion
	if act.LastUpdated != nil && time.Since(*act.LastUpdated) <= defaultUpdateInterval {
return act, nil }
```